Topic: AST

fsvehla opened the pull request:
WIP. Baseline.

jdegoes committed:
Json AST work

fsvehla committed:
Merge remote-tracking branch 'zio/json-ast' into topic/ast

fsvehla committed:
Fix Option.toIterable linting error

fsvehla committed:
Fix existing tests

fsvehla committed:
fmt

calvinlfer committed:
Improve merge so that the array can be reused to produce the final result (#270)

Co-authored-by: calvinlfer <cal@kaizen-solutions.io>

fsvehla committed:
JsonCursor[From, To] (#267)

* Remove unused parameter from encodeJsonStream

* Remove bound

* Bring back bound

* JsonCursor[From, To]

* .field / .element

* prepare

* JsonSpec

* fmt test

* Implement equals with preliminary tests

Co-authored-by: calvinlfer <calvin.l.fer@gmail.com>

calvinlfer committed:
Change objEqual to avoid using chunk.tail and use find (#271)

instead to obtain early exit semantics and quicker iteration

fsvehla committed:
Merge remote-tracking branch 'zio/develop' into topic/ast

calvinlfer committed:
Implement hashCode (consistent with equals) and add tests (#275)

Co-authored-by:  fsvehla <ferdinand@fsvehla.at>

Co-authored-by: cal3 <cal@kaizen-solutions.io>
Co-authored-by: fsvehla <ferdinand@fsvehla.at>

calvinlfer committed:
Implement foldUp (#279)

Add tests for foldUp and foldDown

fsvehla committed:
JsonCursor variance (#280)

* Variance, remove type bound alternative experiment

* final

* Remove comment

jdegoes reviewed (approved):
Excellent work!

jdegoes commented:
Good to merge when you're happy with it. There's a few unimplemented methods still but looking pretty solid.

fsvehla commented:
@jdegoes I’ll have transformation with cursors / delete / transformAt ready in a day or two and will ping you then.

fsvehla committed:
Simplify JsonCursor >>>, field, element (#291)

* More >>> tests

* Move field / element to JsonCursor

This also leads to better compile messages:

Cannot prove that Json <: Json.Obj
vs.
value field is not a member of JsonCursor.Identity

* Remove superfluous JsonCursor.AndThen

fsvehla committed:
Merge, tests (#296)

fsvehla committed:
Merge remote-tracking branch 'zio/develop' into topic/ast

fsvehla committed:
Remove AST methods which are not yet implemented for now

fsvehla committed:
Merge remote-tracking branch 'zio/develop' into topic/ast

# Conflicts:
#	zio-json/jvm/src/main/scala/zio/json/JsonEncoderPlatformSpecific.scala
